src/main/java/com/conveyal/taui/controllers/BrokerController.java

```diff
@@ -175,10 +175,13 @@ private Object singlePoint(Request request, Response response) {
             // probably degrades the perceived responsiveness of single-point requests.
             return ByteStreams.toByteArray(entity.getContent());
         } catch (SocketTimeoutException ste) {
-            LOG.info("Timeout waiting for response from worker. Perhaps an old version of R5 has blocked while preparing a network.");
+            LOG.info("Timeout waiting for response from worker.");
             // Aborting the request might help release resources - we had problems with exhausting connection pools here.
             httpPost.abort();
-            return jsonResponse(response, HttpStatus.ACCEPTED_202, "Preparing network for analysis");
+            return jsonResponse(response, HttpStatus.SERVICE_UNAVAILABLE_503, "Routing server timed out. For the " +
```

Comment:
500 errors should not be manually returned to the browser from the backend. 

While the worker timing out may signal that it's service is unavailable, the backend is still perfectly available and able to handle requests. I would recommend using a 40x HTTP code.